packages/grafana-runtime/src/services/pluginExtensions/usePluginExtensions.ts

```diff
@@ -55,6 +66,17 @@ export function usePluginComponents<Props = {}>(
   );
 }
 
+export function PluginComponentsProvider<Props = {}>({
+  children,
+  options,
+}: {
+  children: (result: ReturnType<typeof usePluginComponents<Props>>) => React.ReactNode;
+  options: GetPluginExtensionsOptions;
+}) {
+  const result = usePluginComponents<Props>(options);
+  return children(result);
+}
+
```

Comment:
Hmm, I am a bit concerned about these (`PluginComponentsProvider` and `PluginLinksProvider`). On one hand these can be useful when working with class components which cannot use hooks - on the other hand this means that you would inline logic in your class components tree that could live in a separate sub-component (that uses the hooks). Using sub-components in these cases is not much more complicated IMHO, while this would introduce _another_ way of accessing extensions.

Reply:
Ah see what you mean. Yeah I was just thinking this as sort of functional vs class component symmetry but yeah making a functional subcomponent would probably be nicer to read anyway, will change.